Dockerfile.windows-build

```diff
@@ -41,24 +42,49 @@ RUN curl --proto '=https' --tlsv1.2 -sSf https://sh.rustup.rs | sh -s -- -y --de
+RUN mkdir -p /root/.cargo && \
+    printf '[target.x86_64-pc-windows-gnullvm]\n\
+linker = "clang"\n\
+rustflags = [\n\
+    "-C", "link-arg=--target=x86_64-w64-mingw32",\n\
+    "-C", "link-arg=--sysroot=/clang64",\n\
+    "-C", "link-arg=-fuse-ld=lld",\n\
+    "-C", "link-arg=--rtlib=compiler-rt",\n\
+    "-C", "link-arg=-resource-dir=/clang64/lib/clang/21",\n\
+    "-C", "link-arg=-L/clang64/lib",\n\
+]\n' > /root/.cargo/config.toml
+
 # Create build script
 RUN printf '#!/bin/bash\n\
 set -e\n\
 \n\
 echo "Building with Clang for Windows..."\n\
 export CC="clang"\n\
-export CFLAGS="--target=x86_64-w64-mingw32 --sysroot=/clang64 -I/clang64/include -fms-extensions -fdeclspec -D_WIN32 -D_WIN32_WINNT=0x0600 -DWIN32 -fuse-ld=lld --rtlib=compiler-rt -resource-dir=/clang64/lib/clang/21"\n\
+export CFLAGS="--target=x86_64-w64-mingw32 --sysroot=/clang64 -I/clang64/include -I/clang64/include/SDL2 -Isrc -Iinclude -fms-extensions -fdeclspec -D_WIN32 -D_WIN32_WINNT=0x0600 -DWIN32 -fuse-ld=lld --rtlib=compiler-rt -resource-dir=/clang64/lib/clang/21"\n\
 export LDFLAGS="-O3 -gdwarf-4 --target=x86_64-w64-mingw32 --sysroot=/clang64 -fuse-ld=lld --rtlib=compiler-rt -resource-dir=/clang64/lib/clang/21 -Wl,-subsystem,windows"\n\
-\n\
+export WINDRES="x86_64-w64-mingw32-windres --preprocessor=clang --preprocessor-arg=--target=x86_64-w64-mingw32 --preprocessor-arg=-E --preprocessor-arg=-xc --preprocessor-arg=-DRC_INVOKED"\n\
 # Create sdl2-config wrapper\n\
-printf '"'"'#!/bin/bash\ncase "$1" in\n  --cflags) echo "-I/clang64/include/SDL2" ;;\n  --libs) echo "-L/clang64/lib -lSDL2main -lSDL2" ;;\n  --prefix) echo "/clang64" ;;\nesac\n'"'"' > /usr/local/bin/sdl2-config\n\
-chmod +x /usr/local/bin/sdl2-config\n\
+mkdir -p /clang64/bin\n\
+printf '"'"'#!/bin/bash\ncase "$1" in\n  --cflags) echo "-I/clang64/include/SDL2" ;;\n  --libs) echo "-L/clang64/lib -lSDL2main -lSDL2" ;;\n  --prefix) echo "/clang64" ;;\nesac\n'"'"' > /clang64/bin/sdl2-config\n\
+chmod +x /clang64/bin/sdl2-config\n\
+# Ensure project-local cargo config exists\n\
+mkdir -p astonia_net/.cargo\n\
+if [ ! -f astonia_net/.cargo/config.toml ]; then\n\
+  printf '"'"'[target.x86_64-pc-windows-gnullvm]\\nlinker = "clang"\\nrustflags = [\\n    "-C", "link-arg=--target=x86_64-w64-mingw32",\\n    "-C", "link-arg=--sysroot=/clang64",\\n    "-C", "link-arg=-fuse-ld=lld",\\n    "-C", "link-arg=--rtlib=compiler-rt",\\n    "-C", "link-arg=-resource-dir=/clang64/lib/clang/21",\\n    "-C", "link-arg=-L/clang64/lib",\\n]\\n'"'"' > astonia_net/.cargo/config.toml\n\
+fi\n\
+\n\
 \n\
 clang --version\n\
-make -f build/make/Makefile.windows CC="$CC" CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS"\n\
+make -f build/make/Makefile.windows CC="$CC" CFLAGS="$CFLAGS" LDFLAGS="$LDFLAGS" WINDRES="$WINDRES"\n\
+\n\
+# Copy Rust DLL to bin directory\n\
+echo "Copying Rust DLL to bin..."\n\
+cp astonia_net/target/x86_64-pc-windows-gnullvm/release/astonia_net.dll bin/ || { echo "Failed to copy DLL"; exit 1; }\n\
```

Comment:
I would also think this should be handled by the Makefile directly, rather than having to be a step here - given we already construct the relative address automatically in the Makefile